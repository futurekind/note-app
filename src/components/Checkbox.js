import React, { PropTypes } from 'react';
import styled from 'styled-components';

import { colors } from '../utils/styles';
import Icon from './Icon';

const View = styled.span`
    width: 24px;
    height: 24px;
    display: inline-block;
    background-color: ${ props => props.background };
    cursor: pointer;
`

const CheckboxIcon = styled(Icon)`
    fill: #fff;
`

const Checkbox = ({
    checked,
    type,
    onCheck
}) => {
    return (
        <View background={ colors[type] } onClick={ () => onCheck(checked) }>
            { checked && <CheckboxIcon id="check" /> }
        </View>
    )
}

Checkbox.defaultProps = {
    onCheck: () => {},
    checked: false
}

Checkbox.propTypes = {
    checked: PropTypes.bool,
    type: PropTypes.oneOf([
        'orange',
        'red',
        'green',
        'blue',
        'purple'
    ]),
    onCheck: PropTypes.func,
}

export default Checkbox