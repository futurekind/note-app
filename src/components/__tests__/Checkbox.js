import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Checkbox from '../Checkbox';

describe('Checkbox', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <Checkbox />
        )
    })

    it('renders', () => {
        expect(toJSON(wrapper)).toMatchSnapshot();
    })

    it('renders bg colors of type prop', () => {
        wrapper.setProps({
            type: 'blue'
        })

        expect(toJSON(wrapper)).toMatchSnapshot()
    })

    it('renders checked Icon', () => {
        wrapper.setProps({
            checked: true
        })

        expect(toJSON(wrapper)).toMatchSnapshot()
    })

    it('calls onCheck callback', () => {
        const spy = jest.fn();

        wrapper.setProps({
            onCheck: spy
        })

        wrapper.simulate('click');

        expect(spy).toHaveBeenCalledWith(false)

        wrapper.setProps({
            checked: true
        })

        wrapper.simulate('click');

        expect(spy).toHaveBeenLastCalledWith(true)
    })

})