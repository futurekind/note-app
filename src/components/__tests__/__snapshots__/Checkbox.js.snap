exports[`Checkbox renders 1`] = `
<styled.span
  background="transparent"
  onClick={[Function]} />
`;

exports[`Checkbox renders bg colors of type prop 1`] = `
<styled.span
  background="#84B2D2"
  onClick={[Function]} />
`;

exports[`Checkbox renders checked Icon 1`] = `
<styled.span
  background="transparent"
  onClick={[Function]}>
  <Styled(Icon)
    fill="#353535"
    id="check" />
</styled.span>
`;
